control-plane/csi-driver/src/bin/controller/controller.rs

```diff
@@ -861,10 +861,19 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
             // If snapshot is already created in previous attempts, fetch and return it.
             Ok(snapshot) => Ok(snapshot),
             Err(ApiClientError::ResourceNotExists(_)) => {
-                if let Some(app_node_endpoint) = app_node_endpoint_info.clone() {
-                    issue_fs_freeze(app_node_endpoint, volume_uuid.to_string()).await?;
+                if let Some(ref app_node_endpoint) = app_node_endpoint_info {
+                    if let Err(error) =
+                        issue_fs_freeze(app_node_endpoint.clone(), volume_uuid.to_string()).await
+                    {
+                        return match error.code() {
+                            Code::NotFound => Err(Status::not_found(format!(
+                                "Failed to freeze volume {}, filesystem volume is not mounted",
+                                volume_uuid
+                            ))),
+                            _ => Err(error),
+                        };
+                    }
```

Comment:
I think only 1 match suffices

```suggestion
                    match issue_fs_freeze(app_node_endpoint.clone(), volume_uuid.to_string()).await
                    {
                        Err(error) if error.code() == Code::NotFound => Err(Status::not_found(format!(
                                "Failed to freeze volume {}, filesystem volume is not mounted",
                                volume_uuid
                            ))),
                        _else => _else
                    }?;
```
Also is the volume not mounted, or not staged/attached?

Reply:
I wanted to use attached, was wondering if users would be more used to the term mounted than attached?

Reply:
hmm not sure. Volume attachments is using the term attached, but they would have to know about it.

Reply:
Alright. I will use attached here.

Reply:
Done.